display/SINK.print/print_callbacks.hpp

```diff
@@ -0,0 +1,1003 @@
+#include <unordered_map>
+#include <tuple>
+#include <set>
+#include <vector>
+#include <cmath>
+#include <iostream>
+#include <iomanip>
+#include <regex>
+#include <iostream>
+
+#include "json.hpp"
+#include "my_demangle.h"
+
+//! Backends identifiers enum.
+
+//! Every backend reaching an lttng tracepoint, that subsequently 
+//! generates a message, is identified by a backend_id. This 
+//! identifies the backend that generated a given message.
+enum backend_e{ 
+    BACKEND_UNKNOWN = 0,
+    BACKEND_ZE = 1,
+    BACKEND_OPENCL = 2,
+    BACKEND_CUDA = 3,
+    BACKEND_OMP_TARGET_OPERATIONS = 4,
+    BACKEND_OMP = 5 
+};
+
+//! Backends identifiers enum.
+
+//! Every backend reaching an lttng tracepoint, that subsequently 
+//! generates a message, is identified by a backend_id. This 
+//! identifies the backend that generated a given message.
+constexpr const char* backend_name[] = { 
+    "BACKEND_UNKNOWN",
+    "BACKEND_ZE",
+    "BACKEND_OPENCL",
+    "BACKEND_CUDA",
+    "BACKEND_OMP_TARGET_OPERATIONS",
+    "BACKEND_OMP" 
+};
+
+//! Backends levels.
+
+//! "backend_level" maps a "backend_id" with its level of abstraction.
+//! For instance, OpenMP in level 0 run on top of LO/OpenCL/CUDA.
+//! This abstraction hierachy is represented by "backend_level".
+constexpr int backend_level[] = { 
+    2, // BACKEND_UNKNOWN
+    2, // BACKEND_ZE
+    2, // BACKEND_OPENCL
+    2, // BACKEND_CUDA
+    1, // BACKEND_OMP_TARGET_OPERATIONS
+    0  // BACKEND_OMP
+};
+
+// TODO: Referenced in THAPI (utils/xprof_utils.hpp), but not used in metababel.
+typedef enum backend_e backend_t;
+
+// Datatypes represeting string classes (or messages) common data. 
+
+typedef intptr_t      process_id_t;
+typedef uintptr_t     thread_id_t;
+typedef std::string   hostname_t;
+typedef std::string   thapi_function_name;
+typedef uintptr_t     thapi_device_id;
+
+// Represent a device and a sub device
+
+// TODO: Referenced in THAPI (opencl/clinterval_callbacks.hpp,utils/xprof_utils.hpp,opencl/clinterval_callbacks.cpp.erb), but not used in metababel, not sure if needed.
+typedef std::tuple<thapi_device_id, thapi_device_id> dsd_t;
+
+// TODO: Not referenced in any other place of THAPI (babeltrace) neither metababel.
+typedef std::tuple<hostname_t, thapi_device_id> h_device_t;
+
+// TODO: Referenced in THAPI (ze/zeinterval_callbacks.hpp,ze/zeinterval_callbacks.cpp.erb,utils/xprof_utils.hpp) but not used in metababel, not sure if needed.
+typedef std::tuple<hostname_t, process_id_t> hp_t;
+
+// TDOD: Referenced in THAPI (cuda/cudainterval_callbacks.cpp.erb,ze/zeinterval_callbacks.hpp,opencl/clinterval_callbacks.hpp,ze/zeinterval_callbacks.cpp.erb,omp/ompinterval_callbacks.hpp,utils/xprof_utils.hpp)
+//! Identifies an entity in a parallel/distributed program that generates a message. 
+typedef std::tuple<hostname_t, process_id_t, thread_id_t> hpt_t;
+
+//! Identifies an specific host, process, thread using the api call "thapi_function_name" in a parallel/distributed program
```

Comment:
please let's not duplicate this code between `aggreg` and `print`

Reply:
Agreed it was a baby "step" now I will put these definitions in a single include file.